src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/PupilBuildingModuleView.java

```diff
@@ -8,8 +8,8 @@
 public class PupilBuildingModuleView extends WorkerBuildingModuleView
 {
     @Override
-    public boolean canAssign(final ICitizenDataView data)
+    public boolean canAssign(final ICitizenDataView citizen)
     {
-        return data.isChild();
+        return citizen.isChild() && (citizen.getWorkBuilding() == null || getAssignedCitizens().contains(citizen.getId()));
```

Comment:
seems a bit weird, we allow assigning a child which is already assigned?

Reply:
Isn’t this to show the children in the manage worker GUI? That gui usually shows the unemployed citizens (children) and already assigned workers.

Reply:
^